noodles-bed/src/ser.rs

```diff
@@ -491,47 +511,100 @@ impl<'a> ser::SerializeStructVariant for &'a mut Serializer {
     }
 }
 
-////////////////////////////////////////////////////////////////////////////////
+#[cfg(test)]
+mod serde_tests {
+    use super::*;
+
+    #[test]
+    fn test_struct() {
+        #[derive(Serialize)]
+        struct Test {
+            int: u32,
+            seq: Vec<&'static str>,
+        }
+
+        let test = Test {
+            int: 1,
+            seq: vec!["a", "b"],
+        };
+        let expected = r#"{"int":1,"seq":["a","b"]}"#;
+        assert_eq!(to_string(&test).unwrap(), expected);
+    }
+
+    #[test]
+    fn test_enum() {
+        #[derive(Serialize)]
+        enum E {
+            Unit,
+            Newtype(u32),
+            Tuple(u32, u32),
+            Struct { a: u32 },
+        }
+
+        let u = E::Unit;
+        let expected = r#""Unit""#;
+        assert_eq!(to_string(&u).unwrap(), expected);
 
-#[test]
-fn test_struct() {
-    #[derive(Serialize)]
-    struct Test {
-        int: u32,
-        seq: Vec<&'static str>,
+        let n = E::Newtype(1);
+        let expected = r#"{"Newtype":1}"#;
+        assert_eq!(to_string(&n).unwrap(), expected);
+
+        let t = E::Tuple(1, 2);
+        let expected = r#"{"Tuple":[1,2]}"#;
+        assert_eq!(to_string(&t).unwrap(), expected);
+
+        let s = E::Struct { a: 1 };
+        let expected = r#"{"Struct":{"a":1}}"#;
+        assert_eq!(to_string(&s).unwrap(), expected);
     }
 
-    let test = Test {
-        int: 1,
-        seq: vec!["a", "b"],
-    };
-    let expected = r#"{"int":1,"seq":["a","b"]}"#;
-    assert_eq!(to_string(&test).unwrap(), expected);
-}
+    #[test]
+    fn test_bed_single_serialization() {
+        // This doesn't sound like the correct place for this test
+        //    (code smell from importing stuff not defined here?)
+        //    (maybe there is an easier way to build).
+        use noodles_core::Position;
+
+        // wait how come am i not testing deserialization aswell
+        let record = Record::<3>::builder()
+            .set_reference_sequence_name("sq0")
+            .set_start_position(Position::try_from(8).expect("Failed to create position"))
```

Comment:
Might as well `.unwrap()`. Would save you from having to write an informative errors.

Reply:
Fixed, I think it's cleaner aswell

This part of the code should hopefully change anyway, to accomodate for better errors (more details on my comment below)